### DISCUSSION TITLE
cat infested

### DISCUSSION THREAD

**Discussion by M1ngM1ngQQY** [ultralytics/ultralytics#7221] [Q&A]
I would like to ask, let's say my model can recognize these three categories (people, cats, dogs), I use a camera to monitor at my door, I detect a cat appearing, how do I get this information. I want to make a cat that appears, and then my UI pops up with "cat infested", how should I do it?

**Comment by pderrenger** [comment]
@M1ngM1ngQQY to achieve the functionality you're describing, you would typically set up a YOLOv8 model to perform object detection on the video feed from your camera. Once the model detects a cat, you can trigger a UI update to display the message "cat infested". Here's a high-level overview of how you might implement this:

1. **Load the Model**: Load your YOLOv8 model that's been trained to recognize people, cats, and dogs.

    ```python
    from ultralytics import YOLO
    model = YOLO('path/to/your/model.pt')
    ```

2. **Perform Detection**: Use the `predict` method to perform detection on frames from your camera feed.

    ```python
    results = model.predict(source='path/to/camera/feed')
    ```

3. **Check for Cats**: After each prediction, check if a cat has been detected.

    ```python
    for result in results:
        if 'cat' in [model.names[i] for i in result.boxes.cls]:
            # Trigger UI update here
            update_ui_with_message("cat infested")
    ```

4. **Update UI**: Implement the `update_ui_with_message` function to interact with your UI and display the message.

    ```python
    def update_ui_with_message(message):
        # Your code to update the UI, this is highly dependent on your UI framework
        print(message)  # As a placeholder, just print the message to the console
    ```

This is a simplified example, and the actual implementation will depend on the specifics of your camera system, UI framework, and how you're processing the video feed. You'll also need to handle threading or asynchronous processing if you want to perform real-time updates without blocking the video feed processing.

For more detailed information on using the `predict` method and handling the results, you can refer to the Predict section in the Ultralytics Docs.
